pkg/performance/collector.go

```diff
@@ -51,11 +52,20 @@ type NewContinuousCollector func(logr.Logger, CollectionConfig) (ContinuousColle
 type NewPointCollector func(logr.Logger, CollectionConfig) (PointCollector, error)
 
 type CollectorCapabilities struct {
-	SupportsOneShot    bool
-	SupportsContinuous bool
-	RequiresRoot       bool
-	RequiresEBPF       bool
-	MinKernelVersion   string
+	SupportsOneShot      bool
+	SupportsContinuous   bool
+	RequiredCapabilities []capabilities.Capability
+	MinKernelVersion     string
+}
+
+// CanRun checks if the collector can run with current process capabilities
+func (c CollectorCapabilities) CanRun() (bool, []capabilities.Capability, error) {
+	if len(c.RequiredCapabilities) == 0 {
+		return true, nil, nil
+	}
+
+	hasAll, missing, err := capabilities.HasAllCapabilities(c.RequiredCapabilities)
+	return hasAll, missing, err
```

Comment:
another way you could do this is define the signature like
```golang
func (c CollectorCapabilities) CanRun() error {}
```

If can run then error is nil, otherwise you can list an error per capability missing:
```
errors.Join(MissingCapability, MissingCapability, ...)
```
https://pkg.go.dev/errors#Join